### ISSUE TITLE
Tooltip creates invalid DOM nesting warnings

### ISSUE THREAD

**Issue by lederer** [raster-foundry/blasterjs#133]
Per https://github.com/raster-foundry/blasterjs/pull/128#pullrequestreview-202913777

> There is a warning about invalid DOM nesting which I'd like to handle at some point in the future with another PR. It could have undesired outcomes in different browsers.
>
> For example, currently, if you apply a tooltip in a button and have a button within that tooltip (see doc example), technically you have illegally nested buttons as far as browsers are concerned. But the way we ultimately render the component removes the nested structure in the DOM but only after the browser gives us a console warning. Since, at the end of the day, we aren't maintaining an illegal nesting structure, I propose we make an issue for this and fix it later.

Solution might involve using a portal. 

Could also be a good opportunity to turn Tooltip into a hook (`useTooltip`). Maybe use it in `Box` so all components get it for free. Tooltip content could be passed in via a `tooltip` prop.
